### DISCUSSION TITLE
Fail to boot on RX 6700 XT

### DISCUSSION THREAD

**Discussion by pmnlla** [ChefKissInc/NootRX#55] [Q&A]
Greetings!

With NootRX, my MacOS install fails to boot. On my 6700XT, I get past the kernel to macOS handoff, and then I get a black screen. No apple logo, not even a displayport signal coming out of the GPU. 

I've tried using adgpmod boot flags, updating NootRX, and I'm using the MacPro7,1 SMBIOS.

Does anyone know what the issue could be?

**Comment by Maxritz** [comment]
I have had the same issue. I updated to 14.5 beta and it kept crashing. I switched back to whatevegreen and disabled nootrx. it works albeit in 7mb framebugger in Sonoma. The nootrx works well on 14.1 and older versions.

**Reply by pmnlla** [reply]
On the 6700XT? Navi 22 isn't supported by WhateverGreen.

**Reply by Maxritz** [reply]
it works without acceleration add agpmod=pikera and unfairvga=1 and you can boot 14.5 Sonoma as well

my flags
-v keepsyms=1 npci=0x3000 debug=0x100 alcid=1 agpmod=pikera unfairvga=1

![Uploading Screenshot 2024-05-12 at 8.25.08 PM.png…]()

**Reply by VisualEhrmanntraut** [reply]
stop misleading the guy into using the card unaccelerated just because you don't care about it

**Reply by pmnlla** [reply]
i dont know how to tell you this but 90% of workloads you ever want to do on a computer require graphics accel, and unfairgva doesn't apply

**Reply by Maxritz** [reply]
my bad. I am on 14.5 and no pending updates too. but i guess i am doing something wrong on my end.

**Reply by pmnlla** [reply]
there's no shame in needing help, but encouraging someone to ignore gpu acceleration kind of breaks things EVERYWHERE. 
check if you've got pending updates with https://github.com/macadmins/nudge

**Reply by Maxritz** [reply]
i want trying to miselad! i was trying to fix my 14.5 and the only way it seems to work was disabling nootrx and enabling whatevergreen . i will try that nudge and see as well. i have been through all the updates as of now from what i see, but maybe i am wrong, will test.

**Comment by pmnlla** [comment]
@VisualEhrmanntraut any ideas?

**Comment by VisualEhrmanntraut** [comment]
https://chefkissinc.github.io/nootrx

**Reply by pmnlla** [reply]
Guessing this is the lilu bug you referred to? Which is weird, there's no pending updates and I got to the desktop on the last successful boot

**Reply by VisualEhrmanntraut** [reply]
read

**Reply by Maxritz** [reply]
did you get the 6700xt working on 14.5 beta? i would be thankful if you shared what you did. It seems that my system crashes with the current nootrx. i tried the latest CI artefact as well

**Reply by pmnlla** [reply]
i gotta learn to stop assuming basesystem comes with the latest updates istg

**Reply by Maxritz** [reply]
so you are on 14.5 like me? or are you on 14.1

**Reply by pmnlla** [reply]
14.4, one minor update behind. Latest basesystem I could get. Gonna update to 14.5.

**Reply by Maxritz** [reply]
well it seems like i was being absolutely stupid. nudge didnt mention i needed any updates. but here is what i have
i have nootrx enabled with the following boot switches.
-v keepsyms=1 npci=0x3000 debug=0x100 alcid=1 agpmod=pikera unfairvga=1
i enabled algrey mtrr as well. will swap and then remove the switches and check later. My bad mate. Thank you all

**Reply by VisualEhrmanntraut** [reply]
the last two boot args don't exist in nootrx

**Reply by pmnlla** [reply]
Small update. I'm on MacOS 14.4.1, the latest version, according to apple. 14.5 seems to be a beta. No idea what I'm missing. I've attached my latest opencore log.
[opencore-2024-05-12-112234.txt](https://github.com/ChefKissInc/NootRX/files/15286814/opencore-2024-05-12-112234.txt)
My config is here: https://bashupload.com/eMz6F/config.plist

**Reply by pmnlla** [reply]
Another update. Tried the iMacPro1,1 SMBIOS, instead of the MacPro7,1. No change.

**Reply by VisualEhrmanntraut** [reply]
why would that help

**Reply by Maxritz** [reply]
on your config, you have used whatever green without any switches. 


As directed by @VisualEhrmanntraut  to the nootrx link which states : 
"Due to a Lilu bug, NootRX cannot be used in recovery or during upgrades due to a misdetection causing a stall at the last stage. For now, have it enabled only after completing install or update."

I fixed my issue as follows:
with that nudge, i checked for pending updates. 
config.plist - unchecked NootRX and enabled whatevergreen kept the mac pro at 7,1 
This needed agpmod=pikera unfairvga=1 to work. I let it finish all the updates etc. Then after a normal boot, disabled whatevergreen and switched NootRX back on .
it seems to work for me.

**Reply by VisualEhrmanntraut** [reply]
you cannot use WhateverGreen with an unsupported GPU. WhateverGreen does nothing if the GPU is not supported

**Reply by VisualEhrmanntraut** [reply]
(so remove WhateverGreen, and just toggle NootRX)

**Reply by Maxritz** [reply]
https://bashupload.com/zTaW4/config.plist

this is my config that seems to work quite well for me on sonoma 14.5. i will add the rest of the kexts later. for now everything works.
@pmnlla  see if this config can help

**Reply by pmnlla** [reply]
> why would that help

On the nootedred page, you said the MacPro7,1 smbios would cause a black screen. I doubt that applies to NootRX, but I ran out of ideas lol. Figured it's worth a try.

**Reply by VisualEhrmanntraut** [reply]
that is because integrated gpu. this obviously does not apply here

**Reply by pmnlla** [reply]
I ran out of options. Do you think you could have a look at my config, to tell me if there's anything obviously wrong? And do you have a telegram I can contact you at? It seems easier to troubleshoot rapidly that way.

**Reply by Maxritz** [reply]
what is the motherboard you are using?

**Reply by Maxritz** [reply]
try my efi and see if it helps. i am on a ryzen 5600 edit the 3 cpu patches 
and the 6700xt . if it works for you , it will be great. remember until patching is complete.. use whatevergreen.. post that switch to nootrx . i still kept the switches. remove the bluetooth stuff if you dont want as well

[EFI Sharing is not allowed, and your GitHub repository is violating copyright laws]

**Reply by pmnlla** [reply]
On a ryzen 5800x3d, with a 6700XT. Using an MSI B450 A-Pro MAX board. Fits the apple naming scheme. I'll try your EFI, since we're running relatively similar hardware.

**Reply by pmnlla** [reply]
Your EFI didn't work for me, probably because of differing ACPI tables. Didnt wanna bother fixing it, i'll probably try remaking my EFI. SSDTTime produced wayyyy more AML files than I previously had, so I was probably missing a whole load of EFI config tables. 🖕 MSI

**Reply by pmnlla** [reply]
Remade the EFI. Now I'm back to square 1.

**Comment by pmnlla** [comment]
Fyi to anyone coming to this, this issue still isn't solved. Just bumping it.

**Comment by pmnlla** [comment]
Closed as this is moved to an issue.
